pkg/serverless/daemon/daemon.go

```diff
@@ -174,49 +218,78 @@ func (d *Daemon) UseAdaptiveFlush(enabled bool) {
 }
 
 // TriggerFlush triggers a flush of the aggregated metrics, traces and logs.
-// They are flushed concurrently.
+// If the flush times out, the daemon will stop waiting for the flush to complete, but the
+// flush may be continued on the next invocation.
```

Comment:
what happen with in progress work? (http request, gzipping, protobuf serialization etc.)

Reply:
Good question. I think it would just pick up where it left off when the environment unfreezes. In the case of an HTTP request though, it is possible that the request timeout would have been reached or the connection closed during the time between invocations. I tested this manually and didn't observe any errors in the timeout scenario.